Revert PR #44

Benbentwo opened the pull request:
## what
* Reverts PR #44 which introduced a bug when using default routing table

## references
* Fixes #45 
* Reopens https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account/issues/30

Benbentwo commented:
/test all

aknysh commented on accepter.tf:
```diff
@@ -64,36 +64,31 @@ locals {
 # Create routes from accepter to requester
 resource "aws_route" "accepter" {
-  count                     = local.accepter_enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
+  count                     = module.this.enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
```
```suggestion
  count                     = local.enabled ? local.accepter_aws_route_table_ids_count * local.requester_cidr_block_associations_count : 0
```

why are you changing from `local.accepter_enabled` to the global enabled?

aknysh commented on accepter.tf:
```diff
@@ -64,36 +64,31 @@ locals {
+    aws_vpc_peering_connection.requester,
   ]
-
-  timeouts {
```
why removing the timeouts?

Benbentwo replied:
accidental replacement

aknysh reviewed (commented):
please see comments

Nuru reviewed (changes requested):
The `aws_subnet_ids` resource has been deprecated, and we should use `aws_subnets` instead. 

Although `aws_route_table` has not been deprecated, we should probably use `aws_route_tables` instead, as it has the cleaner, more modern interface.

We can fix the original bug by looking up the default route table and using it if no route table is found to be associated with the subnet.

Benbentwo commented:
I believe I addressed your comments.  using `aws_subnet_ids` now. 

We we're already using `aws_route_tables` in the PR so nothing changed there. Tested with one using default route table, and one without.

Benbentwo commented:
/test all

Nuru commented on accepter.tf:
```diff
@@ -48,15 +48,17 @@ data "aws_vpc" "accepter" {
+  filter {
+    name   = "vpc-id"
+    values = [local.accepter_vpc_id]
+  }
```
Include [tags](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets#tags)

Nuru commented on accepter.tf:
```diff
@@ -48,46 +48,48 @@ data "aws_vpc" "accepter" {
 
 locals {
-  accepter_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids))), [])
+  accepter_subnet_ids       = try(local.accepter_enabled ? data.aws_subnets.accepter[0].ids : [], [])
```
Apparently I was wrong about sorting data with respect to `data` sources, which I guess are now read during the plan phase. (See what goes into the `count` on line 84.) Please revert, as `distinct` and `sort` add stability to the plans.

Nuru commented on accepter.tf:
```diff
@@ -48,46 +48,48 @@ data "aws_vpc" "accepter" {
-  count     = local.accepter_enabled ? local.accepter_subnet_ids_count : 0
-  provider  = aws.accepter
-  subnet_id = element(local.accepter_subnet_ids, count.index)
+data "aws_route_tables" "accepter" {
+  count    = local.count
+  provider = aws.accepter
+  vpc_id   = local.accepter_vpc_id
```
This is going to add routes to all the route tables in the VPC. I may have said that was a good idea a year ago, but it strikes me as bad/dangerous now. 

I propose that you add a second `aws_route_tables` resource, and use one to filter based on subnet ID and the other to find the default route table. Only add routes to the subnets' route tables, and only add to the main route table if there is a subnet that does not have a route table.

Nuru reviewed (dismissed):
Backtracking on removing `sort` for data from data sources, since it seems to work. See other comments.

Benbentwo commented:
/test all

Benbentwo commented:
/test all

Benbentwo commented:
/test all

aknysh commented on accepter.tf:
```diff
@@ -48,46 +48,70 @@ data "aws_vpc" "accepter" {
+    values = [each.key]
+  }
+}
+# If we had more subnets than routetables, we should update the default.
```
add an empty line between the data sources

aknysh commented on accepter.tf:
```diff
@@ -124,3 +148,8 @@ output "accepter_accept_status" {
+
+output "accepter_route_table_map" {
+  value       = local.accepter_aws_rt_map
+  description = "Accepter VPC subnet to route table map"
```
```suggestion
  description = "Map of accepter VPC subnet IDs to route table IDs"
```

aknysh commented on accepter.tf:
```diff
@@ -124,3 +148,8 @@ output "accepter_accept_status" {
   description = "Accepter VPC peering connection request status"
 }
+
+output "accepter_route_table_map" {
```
```suggestion
output "accepter_subnet_route_table_map" {
```

aknysh commented on accepter.tf:
```diff
@@ -48,46 +48,70 @@ data "aws_vpc" "accepter" {
 
 locals {
-  accepter_subnet_ids       = try(distinct(sort(flatten(data.aws_subnet_ids.accepter.*.ids))), [])
+  accepter_subnet_ids       = try(local.accepter_enabled ? data.aws_subnets.accepter[0].ids : [], [])
```
do we need `try` here?
it's useful in some cases, but mostly it hides errors that we don't want to hide

Benbentwo replied:
I agree, was thinking of that when writing it. kept in, but there's no need, it's removed now

aknysh reviewed (commented):
@Benbentwo a few nitpicks

Nuru commented:
/test all

Nuru commented on accepter.tf:
```diff
@@ -55,25 +55,48 @@ data "aws_subnets" "accepter" {
     name   = "vpc-id"
     values = [local.accepter_vpc_id]
   }
+  dynamic "filter" {
+    for_each = var.accepter_subnet_tags
+    content {
+      name   = "tag:${filter.key}"
+      values = [filter.value]
+    }
+  }
```
```suggestion
  tags = var.accepter_subnet_tags
```
See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets#tags

Benbentwo replied:
totally misread that documentation

Nuru commented on accepter.tf:
```diff
@@ -126,3 +149,8 @@ output "accepter_accept_status" {
+output "accepter_subnet_route_table_map" {
+  value       = local.accepter_aws_rt_map
+  description = "Map of accepter VPC subnet IDs to route table IDs"
+}
```
```suggestion
}

```

Nuru commented:
/test all
